### DISCUSSION TITLE
[Prettier Plugin] Prettier Plugins doesn't sort any classes on PHP file

### DISCUSSION THREAD

**Discussion by MaideAkdede** [tailwindlabs/tailwindcss#10434] [Help]
I tried to open up an issue on the [Github repository of Tailwindlabs : Prettier-plugin-tailwindcss](https://github.com/tailwindlabs/prettier-plugin-tailwindcss) and got sent here.

Like mentioned in the title, this [prettier-plugin-tailwindcss](https://tailwindcss.com/blog/automatic-class-sorting-with-prettier) doesn't work in any of the **PHP files** I'm working on.

The plugin works perfectly in **css, scss, html, js** files but whenever I'm running it in a **php files** only the regular Prettier Plugin is applied and none of the classes get sorted.

My question is:  **what am I doing wrong? Did I miss something about this plugin not supposed to work in PHP files?**

I've also asked this question on [StackOverflow](https://stackoverflow.com/questions/75181706/why-prettier-plugin-tailwindcss-does-not-sort-my-classes-on-my-wordpress-project) but haven't got any answers yet.

**Local Environment:** 
- VsCode v1.74.3
- macOS Ventura 13.0.1 / Window 10
- WordPress v6.1.1
- Prettier v2.8.3
- prettier-plugin-tailwindcss v0.2.1

**Comment by arisac** [comment]
1. Install `@prettier/plugin-php` https://github.com/prettier/plugin-php
2. Config prettier for `@pretter/plugin-php` and `prettier-plugin-tailwindcss` 👇

###  Using `.prettierrc`
```
{
  // ..
  "plugins": [
    "@prettier/plugin-php",
    "prettier-plugin-tailwindcss" 
  ],
}
```

### Or using `prettier.config.js`
```
module.exports = {
  // ..
  plugins: [
    require('@prettier/plugin-php'),
    require('prettier-plugin-tailwindcss')
  ],
}
```

3. Close and relaunch VS Code, it should be working when you use `>Format Document`

**Reply by matteoFeduzi** [reply]
Hi @arisac, I tried what you wrote, but it doesn't work. Does it work for anyone?

**Comment by matteoFeduzi** [comment]
@arisac's answer didn't work for me and I solved it by using this plugin:
https://marketplace.visualstudio.com/items?itemName=heybourn.headwind

**Comment by runtime-studio-au** [comment]
For anyone still landing here: the plugin can't work on PHP templates because they can't be parsed into a single tree, so I built a standalone CLI that lexes the file instead — finds the PHP islands, then sorts only the `class="..."` values using the official `prettier-plugin-tailwindcss/sorter` engine. No PHP reformatting, handles islands inside class attributes.

[`@runtimestudio/tailwind-sort-php`](https://github.com/runtime-studio-au/tailwind-sort-php) — fuller explanation in https://github.com/tailwindlabs/tailwindcss/discussions/17406#discussioncomment-XXXXXXX (disclosure: I'm the author).
